### PR TITLE
TYP: aggregations.pyx

### PR DESCRIPTION
**Pull request by jbrockmendel**
@mroeschke the type:ignores added in ewm.py look like potential bugs. any thoughts?

### CODE REVIEW

File: pandas/core/window/ewm.py [hunk 1]
```diff
@@ -518,7 +518,9 @@ def cov_func(x, y):
                 x_array,
                 start,
                 end,
-                self.min_periods,
+                # error: Argument 4 to "ewmcov" has incompatible type
+                # "Optional[int]"; expected "int"
+                self.min_periods,  # type: ignore[arg-type]
```
Comment (mroeschke) [inline pandas/core/window/ewm.py:523]: The EWM subclass ensures this is set to `int` before passing to the parent class which has `Optional[int]` typing

**Review by mroeschke — APPROVED** [review]

File: pandas/_libs/window/aggregations.pyi [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    quantile: float,  # float64_t
+    interpolation: str,
+) -> np.ndarray: ...  # np.ndarray[float]
+
+def roll_apply(
+    obj: object,
+    start: np.ndarray,  # np.ndarray[np.int64]
+    end: np.ndarray,  # np.ndarray[np.int64]
+    minp: int,  # int64_t
+    function: object,
```
Comment (simonjayhawkins) [inline pandas/_libs/window/aggregations.pyi:74]: callable? Callable[..., Any] used in _generate_cython_apply_func

File: pandas/_libs/window/aggregations.pyi [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    raw: bool,
+    args: tuple,
+    kwargs: dict,
+) -> np.ndarray: ...  # np.ndarray[float]
```
Comment (simonjayhawkins) [inline pandas/_libs/window/aggregations.pyi:78]: the code starts with

```
if n == 0:
        return obj
```
so the return type could be same as `obj`

File: pandas/_libs/window/aggregations.pyi [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+def roll_apply(
+    obj: object,
+    start: np.ndarray,  # np.ndarray[np.int64]
+    end: np.ndarray,  # np.ndarray[np.int64]
+    minp: int,  # int64_t
+    function: object,
+    raw: bool,
+    args: tuple,
+    kwargs: dict,
```
Comment (simonjayhawkins) [inline pandas/_libs/window/aggregations.pyi:77]: These are passed to func with no restrictions so I think OK to type as

```suggestion
    args: tuple[Any, ...],
    kwargs: dict[str, Any],
```

to be consistent with _generate_cython_apply_func

File: pandas/core/window/ewm.py [hunk 5]
```diff
@@ -584,12 +586,14 @@ def _cov(X, Y):
                     X,
                     start,
                     end,
-                    self.min_periods,
+                    # error: Argument 4 to "ewmcov" has incompatible type
+                    # "Optional[int]"; expected "int"
+                    self.min_periods,  # type: ignore[arg-type]
```
Comment (simonjayhawkins) [inline pandas/core/window/ewm.py:591]: in the outer scope we have 

```
            min_periods = (
                self.min_periods
                if self.min_periods is not None
                else window_indexer.window_size
            )
```
can we not just pass min_periods here.

**Review by simonjayhawkins — COMMENTED** [review]
Thanks @jbrockmendel generally lgtm

File: pandas/core/window/ewm.py (hunk 5 above)
Comment (jbrockmendel) [reply pandas/core/window/ewm.py:591]: cc @mroeschke ?

Comment (mroeschke) [reply pandas/core/window/ewm.py:591]: True; since the EWM constructor always has an numeric `min_periods` this can just be `self.min_periods`

**Review by simonjayhawkins — APPROVED** [review]

**Comment by simonjayhawkins** [comment]
Thanks @jbrockmendel lgtm pending green.

**Comment by jbrockmendel** [comment]
updated + green

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**